buildSrc/src/main/kotlin/buildsrc/conventions/maven-publish.gradle.kts

```
@@ -4,30 +4,41 @@ plugins {
     id("buildsrc.conventions.base")
     `maven-publish`
     signing
+    // NOTE: external plugin version is specified in implementation dependency artifact of the project's build file
+    id("io.github.gradle-nexus.publish-plugin")
 }
 
-
-val signingKeyId = providers.gradleProperty("signingKeyId")
+val signingKey = providers.gradleProperty("signingKeyB64")
 val signingPassword = providers.gradleProperty("signingPassword")
-val signingSecretKeyRingFile = providers.gradleProperty("signingSecretKeyRingFile")
+//val signingSecretKeyRingFile = providers.gradleProperty("signingSecretKeyRingFile")
 
 val signingPropertiesPresent = provider {
-    signingKeyId.isPresent && signingPassword.isPresent && signingSecretKeyRingFile.isPresent
+    signingKey.isPresent && signingPassword.isPresent
 }
 
 val ossrhUsername = providers.gradleProperty("ossrhUsername")
+
 val ossrhPassword = providers.gradleProperty("ossrhPassword")
 
 val isSnapshotVersion = provider { version.toString().endsWith("SNAPSHOT") }
 
-
 val javadocJarStub by tasks.creating(Jar::class) {
     group = JavaBasePlugin.DOCUMENTATION_GROUP
     description = "Stub empty javadoc.jar artifact (a Javadoc jar is required by Maven Central)"
     archiveClassifier.set("javadoc")
 }
 
-
+nexusPublishing {
+    repositories {
+        sonatype {
+            // This was a guess, trying without
+            // this.stagingProfileId.set("io.kweb")
+            username.set(ossrhUsername.get())
+            password.set(ossrhPassword.get())
```

using `.get()` on a provider causes Gradle to immediately fetch the value. On my machine, I don't have the properties defined, so Gradle throws an error during the configuration phase, so I can't load the Gradle config.

Since `username` and `password` are `Property<String>`, you can just pass a `Provider<String>` to them. Then the plugin will fetch them in the execution phase, only if required. That means that I can load the project, and it will only fail if the plugin tries to fetch the values.

```suggestion
            username.set(ossrhUsername)
            password.set(ossrhPassword)
```

https://docs.gradle.org/current/userguide/lazy_configuration.html